xarray/plot/plot.py

```diff
@@ -453,22 +322,238 @@ def __call__(self, **kwargs):
     def hist(self, ax=None, **kwargs):
         return hist(self._da, ax=ax, **kwargs)
 
-    @functools.wraps(line)
-    def line(self, *args, **kwargs):
-        return line(self._da, *args, **kwargs)
-
     @functools.wraps(step)
     def step(self, *args, **kwargs):
         return step(self._da, *args, **kwargs)
 
 
-def override_signature(f):
-    def wrapper(func):
-        func.__wrapped__ = f
+def _plot1d(plotfunc):
+    """
+    Decorator for common 2d plotting logic
 
-        return func
+    Also adds the 2d plot method to class _PlotMethods
+    """
+    commondoc = """
+    Parameters
+    ----------
+    darray : DataArray
+        Must be 2 dimensional, unless creating faceted plots
+    x : string, optional
+        Coordinate for x axis. If None use darray.dims[1]
+    y : string, optional
+        Coordinate for y axis. If None use darray.dims[0]
+    hue : string, optional
+        Dimension or coordinate for which you want multiple lines plotted.
+    figsize : tuple, optional
+        A tuple (width, height) of the figure in inches.
+        Mutually exclusive with ``size`` and ``ax``.
+    aspect : scalar, optional
+        Aspect ratio of plot, so that ``aspect * size`` gives the width in
+        inches. Only used if a ``size`` is provided.
+    size : scalar, optional
+        If provided, create a new figure for the plot with the given size.
+        Height (in inches) of each plot. See also: ``aspect``.
+    ax : matplotlib.axes.Axes, optional
+        Axis on which to plot this figure. By default, use the current axis.
+        Mutually exclusive with ``size`` and ``figsize``.
+    row : string, optional
+        If passed, make row faceted plots on this dimension name
+    col : string, optional
+        If passed, make column faceted plots on this dimension name
+    col_wrap : int, optional
+        Use together with ``col`` to wrap faceted plots
+    xscale, yscale : 'linear', 'symlog', 'log', 'logit', optional
+        Specifies scaling for the x- and y-axes respectively
+    xticks, yticks : Specify tick locations for x- and y-axes
+    xlim, ylim : Specify x- and y-axes limits
+    xincrease : None, True, or False, optional
+        Should the values on the x axes be increasing from left to right?
+        if None, use the default for the matplotlib function.
+    yincrease : None, True, or False, optional
+        Should the values on the y axes be increasing from top to bottom?
+        if None, use the default for the matplotlib function.
+    add_labels : bool, optional
+        Use xarray metadata to label axes
+    subplot_kws : dict, optional
+        Dictionary of keyword arguments for matplotlib subplots. Only used
+        for 2D and FacetGrid plots.
```

Comment:
```suggestion
    subplot_kws : dict, optional
        Dictionary of keyword arguments for matplotlib subplots. Only used
        for FacetGrid plots.
```
2d should not be applicable here right?